### PR TITLE
Refactor regression tests to be their own functions and general improvements

### PR DESCRIPTION
**Pull request by trevorbernard**
Add the ability to run individual tests.

`./test` runs all the tests
`./test test_best_canonical_tip test_indexer_cli_reports` would only run those two tests

### CODE REVIEW

**Review by jenr24G — APPROVED** [review]
good stuff. maybe we can start splitting test cases off into their own files after this

File: test [hunk 1]
```diff
@@ -3,135 +3,154 @@ set -ex
...
-    rm -rf $TMPDIR
-}
-trap cleanup EXIT
-
-cargo build
-
-teardown() {
-    rm -rf $HOME/.mina-indexer/database || true
-    rm -rf $HOME/.mina-indexer/logs || true
+    rm -rf $TMPDIR 2>/dev/null
```
Comment (robinbb) [inline test:12]: Could be disastrous if $TMPDIR is a directory with a space character in any of the directory components. Recommend surrounding by double quotes. Safer still is "${TMPDIR}".

File: test [hunk 2]
```diff
@@ -3,135 +3,154 @@ set -ex
...
+setup() {
+    echo "setup"
 }
 
-client() {
-    ./target/debug/mina-indexer client "$@"
+teardown() {
+    echo "Cleaning up and killing the background indexer with PID: $idxr_pid"
+    kill $idxr_pid 2>/dev/null
+    rm -rf $HOME/.mina-indexer/database || true
```
Comment (robinbb) [inline test:45]: Same comment as above applies with this line and the next one.

**Review by robinbb — DISMISSED** [review]
Approved based on C4 rules, but see my comments about use of 'rm -fr' on an unquoted variable.

File: test [hunk 3]
```diff
@@ -3,135 +3,155 @@ set -ex
...
-    rm -rf $TMPDIR
-}
-trap cleanup EXIT
-
-cargo build
-
-teardown() {
-    rm -rf $HOME/.mina-indexer/database || true
-    rm -rf $HOME/.mina-indexer/logs || true
+    rm -rf ${TMPDIR} 2>/dev/null
```
Comment (robinbb) [inline test:12]: To be clear, it's the surrounding double quotes, not the surrounding braces, that save you from having the TMPDIR being interpreted as multiple directories. So, this is still not safe, FYI.

**Review by robinbb — APPROVED** [review]

**Pull request merged**